### PR TITLE
Adding new data structure

### PR DESCRIPTION
**Pull request by omarmahamid**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by omarmahamid** [comment]
this a data structure that I implement in my university (you can see it in my github https://github.com/omarmahamid/DS)

**Comment by omarmahamid** [comment]
@siriak @yanglbme

File: src/main/java/com/thealgorithms/datastructures/heaps/FibonacciHeap.java [hunk 1]
```diff
@@ -0,0 +1,436 @@
+package com.thealgorithms.datastructures.heaps;
+
+
+public class FibonacciHeap {
+    private static final double GOLDAN_RATIO = (1+Math.sqrt(5))/2;
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/heaps/FibonacciHeap.java:5]:
```suggestion
    private static final double GOLDEN_RATIO = (1+Math.sqrt(5))/2;
```

File: src/main/java/com/thealgorithms/datastructures/heaps/FibonacciHeap.java [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+        HeapNode node = fibonacciHeap.insert(3);
+        fibonacciHeap.insert(1);
+        fibonacciHeap.insert(18);
+        fibonacciHeap.insert(33);
+
+        System.out.println(fibonacciHeap.findMin().getKey());
+        fibonacciHeap.deleteMin();
+        System.out.println(fibonacciHeap.findMin().getKey());
+
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/heaps/FibonacciHeap.java:435]: Please replace this with tests using `@Test`, see examples in the codebase in the `src/test` directory

File: src/main/java/com/thealgorithms/datastructures/heaps/FibonacciHeap.java [hunk 3]
```diff
@@ -0,0 +1,436 @@
+package com.thealgorithms.datastructures.heaps;
+
+
+public class FibonacciHeap {
+    private static final double GOLDAN_RATIO = (1+Math.sqrt(5))/2;
+	private HeapNode min;
+	private static int totalLinks = 0;
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/heaps/FibonacciHeap.java:7]: Please replace tabs with spaces

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/datastructures/heaps/FibonacciHeap.java [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+	 * a constructor for a Heap with one element
+	 * set the min to be the HeapNode with the given key
+	 * @pre key>=0
+	 * @post empty == false
+	 */
+	public FibonacciHeap(int key) {
+		this.min = new HeapNode(key);
+		this.numOfTrees++;
+		this.numOfHeapNodes++;
+	}
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/heaps/FibonacciHeap.java:31]: Tabs

Comment (omarmahamid) [reply src/main/java/com/thealgorithms/datastructures/heaps/FibonacciHeap.java]: done

**Review by siriak — APPROVED** [review]
LGTM, thanks!

**Pull request merged**